models/vrp.rb

```diff
@@ -516,8 +516,8 @@ def self.deduce_unavailable_days(hash, element, start_index, end_index, type)
     def self.deduce_possible_days(hash, element, start_index)
       convert_possible_dates_into_indices(element, hash, start_index)
 
-      element[:first_possible_days] = element[:first_possible_day_indices].to_a.slice(0..(element[:visits_number] || 1) - 1)
-      element[:last_possible_days] = element[:last_possible_day_indices].to_a.slice(0..(element[:visits_number] || 1) - 1)
+      element[:first_possible_days] ||= element[:first_possible_day_indices].to_a.slice(0..(element[:visits_number] || 1) - 1)
+      element[:last_possible_days] ||= element[:last_possible_day_indices].to_a.slice(0..(element[:visits_number] || 1) - 1)
```

Comment:
Is this related to [Marshal dump repetitions](https://github.com/Mapotempo/optimizer-api/pull/353/files#)? It looks like a proper bug fix. (Not asking for you to separate by the way, just keep it in mind for the future that if the fix is separate, it doesn't wait for the independent refacto and we can merge it faster)

Reply:
What if I dump the vrp via `as_json`, make a change (like modify `visits_number`) and then call `vrp.create` this would leave the incorrect `:first_possible_day_indices` in the new object. As long as we don't have a way to mark an object "dirty/clean" I am afraid the best way is to recalculate it -- considering the fact that it is a very rudimentary calculation, it won't have any performance effects. What do you say?